src/test/java/net/sf/jabref/logic/search/SearchQueryTest.java

```diff
@@ -66,4 +66,51 @@ public void testSearchingForOpenBraketInBooktitle() {
         assertTrue(searchQuery.isMatch(e));
     }
 
+    @Test
+    public void testSearchMatchesSingleKeywordNotPart() {
+        BibEntry e = new BibEntry(IdGenerator.next(), BibtexEntryTypes.INPROCEEDINGS.getName());
+        e.setField("keywords", "banana, pineapple, orange");
```

Comment:
It will work for what you write. Hence, `ö` will match `ö`, but not `\"{o}`, and vice versa. Once the LaTeX free BibEntry is functioning it should work both ways.